### PR TITLE
first working version

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+const _ = require('lodash');
+const moment = require('moment-timezone');
+const defaultOptions = {
+  // the date the embargo will be lifted:
+  embargoEnd: `${moment().date()} 01:00:00 PST`,
```
Comment (jgallen23) [inline index.js]: default this to empty and error if its not set

File: index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  server.log(['hapi-embargo', 'info'], `Embargo will be lifted at ${options.embargoEndTime}, time is now ${moment().local()}`);
+  server.ext({
+    type: 'onRequest',
+    method: (request, reply) => {
+      // if bypass option matches then let them pass:
+      if (options.bypass && request.query.bypass === options.bypass) {
+        return reply.continue();
+      }
+      // if it's not a view then let them pass:
+      if (options.viewsOnly && !reply.render) {
```
Comment (jgallen23) [inline index.js:29]: I think this is a better check: `request.response.variety === 'view'`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "author": "First+Third",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/firstandthird/hapi-embargo/issues"
+  },
+  "homepage": "https://github.com/firstandthird/hapi-embargo#readme",
+  "dependencies": {
+    "lodash": "^4.15.0",
+    "logr": "^1.13.0",
```
Comment (jgallen23) [inline package.json]: logr used?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+const _ = require('lodash');
```
Comment (jgallen23) [inline index.js]: just using defaults right?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+exports.register = (server, config, next) => {
+  const options = defaults(config, defaultOptions);
+  if (!options.embargoEnd) {
+    return next('You must specify a time for the embargo to end by passing the embargoEnd parameter');
+  }
+  options.embargoEndTime = moment(options.embargoEnd);
+  // log time that the embargo will be lifted:
+  server.log(['hapi-embargo', 'info'], `Embargo will be lifted at ${options.embargoEndTime}, time is now ${moment().local()}`);
+  server.ext({
+    type: 'onPostHandler',
```
Comment (jgallen23) [inline index.js]: can we do `onPreHandler` ? if not, lets go back to how you originally had it.

**Review by jgallen23 — CHANGES_REQUESTED** [review]

**Comment by jgallen23** [comment]
travis failed too

File: index.js (hunk 5 above)
Comment (orthagonal) [reply index.js]: I tried that originally.  You would think it would work, but it does not.

**Pull request merged**